### PR TITLE
Add project() to sqlite3_cmakelists.txt

### PR DESCRIPTION
**Pull request by brawner**
Attempting to address issue #323. I'm not sure if this is the correct way of going about this since it's being included from another cmakelists.txt. I'm also open to naming the project whatever you all think is appropriate. I just didn't want to name it sqlite3_vendor (already taken) nor sqlite3 (may conflict with another cmake project?). 

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

**Comment by brawner** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9646/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5331/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7873/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9541/)

**Comment by Karsten1987** [comment]
You repos file is incorrect I believe:
```
  ros2/rosbag2:
    type: git
    url: git@github.com:brawner/rosbag2
    version: sqlite3_cmakelists
```

the url should be provided in form of `https://github.com/...`

**Comment by Karsten1987** [comment]
I actually think naming it `sqlite3` is okay as it should not conflict with any CMake context in this case.

**Comment by brawner** [comment]
Arg, thought I could be clever and just use vcs export

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9648/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5332/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7874/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9542/)

File: sqlite3_vendor/sqlite3_cmakelists.txt [hunk 1]
```diff
@@ -8,4 +9,4 @@ install(TARGETS sqlite3
         LIBRARY DESTINATION lib
         ARCHIVE DESTINATION lib
         RUNTIME DESTINATION bin)
-install(FILES sqlite3.h sqlite3ext.h DESTINATION include)
\ No newline at end of file
+install(FILES sqlite3.h sqlite3ext.h DESTINATION include)
```
Comment (Karsten1987) [inline sqlite3_vendor/sqlite3_cmakelists.txt]: I can't quite see it, but what's the diff here?

Comment (brawner) [reply sqlite3_vendor/sqlite3_cmakelists.txt]: A newline was added at the end of the file (my editor does this automatically). Let me know if this needs to be changed back

**Comment by brawner** [comment]
Looks like Dirk got the fix in for osrf_testing_tools. Trying again

Windows [Build Status](http://ci.ros2.org/job/ci_windows/9557/)

File: sqlite3_vendor/sqlite3_cmakelists.txt (hunk 1 above)
Comment (Karsten1987) [reply sqlite3_vendor/sqlite3_cmakelists.txt]: it's all good. I was just wondering as GitHub doesn't really seem to render this change.

**Comment by Karsten1987** [comment]
Does it make sense to run the windows CI job without cyclone and connext? I think these CI failures are addressed more appropriately at another place and shouldn't really block to move this PR further.

**Comment by brawner** [comment]
We've temporarily set buildtools back to 16.4.5. Trying on windows again.

[Build Status](https://ci.ros2.org/job/ci_windows/9569/)

**Comment by brawner** [comment]
Building on windows finally passed. Failing test is a known issue (https://github.com/ros2/rosbag2/issues/305)

**Pull request merged**